e2e/setupIpfsCluster/index.ts

```diff
@@ -0,0 +1,53 @@
+import { execFile, execFileSync, ChildProcess } from 'child_process';
+import path from 'path';
+import waitOn from 'wait-on';
+
+/**
+ * Spawn local ipfs cluster
+ *
+ * @returns uri of the cluster root
+ */
+export async function spawnIpfsCluster() {
+  shutdownNodes();
+
+  const cluster = execFile(
+    'docker',
+    ['compose', 'up', '--force-recreate', 'cluster_proxy'],
+    { cwd: path.join(__dirname) },
+    (error) => {
+      if (error) {
+        throw error;
+      }
+    }
+  );
+
+  try {
+    await waitOn({
+      // https://ipfscluster.io/documentation/reference/api/
+      resources: ['http-get://localhost:8080/id'],
+      delay: 5000,
+      // if your internet connection is slow try to increase timeout to be able to download all images
```

Comment:
I think this line isn't relevant because `localhost` calls don't go over the internet
```suggestion
```

Reply:
Actually the delay in spinning cluster can be caused by internet calls. Cluster need to download `nginx` and `ipfs` images listed in docker-compose

Reply:
Ah gotcha, I see what you mean by `images` now